vendor/assets/javascripts/initializers/asset_uploads.js

```diff
@@ -22,6 +22,10 @@ $(document).on('cmsify:load', function() {
         $(clone).first().removeClass('uk-hidden');
         UIkit.modal('#add-new-' + resourceName).hide();
         $('.js-remove-asset').removeClass('uk-hidden');
+        if (resourceName === 'download') {
+          var urlArray = res.asset.url.split('/');
+          $('#resource_item_name').val(urlArray[urlArray.length - 1]);
+        }
```

Comment:
Anything that happens upon selection of an asset needs to happen only in the `Cmsify.SelectableAsset` class. `Cmsify.removeUpload` is already setting the `data-asset-url`, so when you select an asset, call the `getFileName()` like you are doing above. The only thing that should happen in this callback are things specific to uploading and not selecting.